packages/google_maps_flutter/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -26,6 +26,15 @@ - (instancetype)initMarkerWithPosition:(CLLocationCoordinate2D)position
   }
   return self;
 }
+- (void)showInfoWIndow {
+  _mapView.selectedMarker = _marker;
+}
+- (void)hideInfoWIndow {
+  _mapView.selectedMarker = nil;
```

Comment:
We can check if the selected Maker is the marker that user wants to hide before hiding.
```dart
if (_mapView. selectedMarker == _marker) {
  _mapView.selectedMarker = nil;
}

```